src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -168,6 +177,14 @@ public bool CanRetry
 
                 await WriteTwoBytesAsync((byte)'\r', (byte)'\n', cancellationToken).ConfigureAwait(false);
             }
+
+            if (cookies != null)
+            {
+                await WriteAsciiStringAsync(HttpKnownHeaderNames.Cookie, cancellationToken).ConfigureAwait(false);
+                await WriteTwoBytesAsync((byte)':', (byte)' ', cancellationToken).ConfigureAwait(false);
```

Comment:
Can we use a cached byte[] representation of "Cookie: " and just blit that out with WriteBytesAsync?

Reply:
At some point I'd like to do this for all headers.  We have already resolved them to well-known header values internally, it's just a matter of exposing an enumerator of KVP<HeaderDescriptor, IEnumerable<string>> instead of KVP<string, IEnumerable<string>>.  Given this I don't think we should special case Cookie now.